### PR TITLE
fix(roundrobin): don't retry a node again

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/balancer/roundrobin.lua [hunk 1]
```diff
@@ -16,16 +16,50 @@
...
+
+            return server
+        end,
+        after_balance = function (ctx, before_retry)
+            if not before_retry then
+                return nil
+            end
+
+            if not ctx.balancer_tried_servers then
+                ctx.balancer_tried_servers = {}
```
[the remark this answers is not in the document]
Comment (spacewander) [reply apisix/balancer/roundrobin.lua]: @membphis 
Solved.

File: apisix/balancer/roundrobin.lua [hunk 2]
```diff
@@ -16,16 +16,55 @@
...
+                        break
+                    end
+                else
+                    break
+                end
+            end
+
+            return server
+        end,
+        after_balance = function (ctx, before_retry)
```
Comment (tokers) [inline apisix/balancer/roundrobin.lua:53]: It seems the `before_retry` is always `true` when `after_balance` is called?

Comment (spacewander) [reply apisix/balancer/roundrobin.lua:53]: There is another call in `apisix/init.lua`.

**Review by membphis — APPROVED** [review]

File: apisix/balancer/roundrobin.lua (hunk 2 above)
Comment (tokers) [reply apisix/balancer/roundrobin.lua:53]: got it.

**Review by tokers — APPROVED** [review]

**Pull request merged**